projects/ngx-darkbox-gallery/src/lib/ngx-darkbox-gallery.component.html

```diff
@@ -1,10 +1,21 @@
 <div [ngClass]="getImageGridClasses()" [ngStyle]="{'justify-content': getContentJustification()}">
-  <div class="image-container" [ngStyle]="{'height': getThumbnailHeight(), 'width': getThumbnailWidth()}" *ngFor="let image of images | slice:0:imageCount" (click)="onImageClicked(image)">
-    <img src="{{image.thumbnailUrl}}" alt="{{image.altText}}">
+  <div class="image-container" [ngStyle]="{'height': getThumbnailHeight(), 'width': getThumbnailWidth()}"
+    *ngFor="let image of images | slice:0:imageCount" (click)="onImageClicked(image)">
+
+    <!-- The actual thumbnail image -->
+    <img [ngStyle]="{'display': isLoadingPlaceholderEnabled() && !batchThumbnailsLoaded ? 'none' : ''}"
+      src="{{image.thumbnailUrl}}" alt="{{image.altText}}" (load)="onThumbnailLoaded(image)">
+
+    <!-- Loading placeholder in the configured size -->
+    <loading-placeholder *ngIf="isLoadingPlaceholderEnabled() && !batchThumbnailsLoaded"
```

Comment:
Maybe we can find a cool solution to optionally enable the placeholder for:

- Alle images if new images are loaded (like it is implemented now)
- Only use the placeholder for the new batch until all images of the batch are loaded
- Use the placeholder only as long as the real thumbnail is not available and show the real image as soon as it arrives

Reply:
I tried diffent styles myself (option one and three to be precise), but the implementation for each style differs significantly. Personally I prefer the style now implemented, because in the other varients the thumbnails "jump around" a lot while loading.
I would suggest to implement different styles of the placeholders incrementally if we decide to offer them.

Reply:
For the fluid layout that is true. In the static layout the thumbnails should not jump around because the size of the thumbnails and placeholders are identical. An incremental implementation is fine for me.